http/server.go

```diff
@@ -0,0 +1,132 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ipfs/go-log/v2"
+
+	"github.com/drand/drand/beacon"
+	"github.com/drand/drand/core"
+	"github.com/drand/drand/key"
+	"github.com/drand/drand/protobuf/drand"
+)
+
+// New creates an HTTP handler for the public Drand API
+func New(ctx context.Context, client drand.PublicClient) (http.Handler, error) {
+	pkt, err := client.Group(ctx, &drand.GroupRequest{})
+	if err != nil {
+		return nil, err
+	}
+	if pkt == nil {
+		return nil, fmt.Errorf("Failed to retrieve valid GroupPacket")
+	}
+	parsedPkt, err := core.ProtoToGroup(pkt)
+	if err != nil {
+		return nil, err
+	}
+
+	handler := handler{ctx, client, parsedPkt, log.Logger("http")}
+
+	mux := http.NewServeMux()
+	//TODO: aggregated bulk round responses.
+	mux.HandleFunc("/public/latest", handler.LatestRand)
+	mux.HandleFunc("/public/", handler.PublicRand)
+	mux.HandleFunc("/group", handler.Group)
+	return mux, nil
+}
+
+type handler struct {
+	ctx       context.Context
+	client    drand.PublicClient
+	groupInfo *key.Group
+	log       log.StandardLogger
+}
+
+func (h *handler) getRand(round uint64) ([]byte, error) {
+	req := drand.PublicRandRequest{Round: round}
+	resp, err := h.client.PublicRand(h.ctx, &req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return json.Marshal(resp)
+}
+
+func (h *handler) PublicRand(w http.ResponseWriter, r *http.Request) {
+	// Get the round.
+	round := strings.Replace(r.URL.Path, "/public/", "", 1)
```

Comment:
Wouldn't be more robust / safe to rely on Go's native method for this ? https://golangbyexample.com/net-http-package-get-query-params-golang/

Reply:
this switched from using query parameters "/public?round=n" to using the path directly per @hsanjuan 's ask: "/public/n"

Reply:
Oh right sorry - it's all good